epam-star-wars-heroes-wiki/epam-star-wars-heroes-wiki/UI/Search/ViewController.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  ViewController.swift
+//  epam-star-wars-heroes-wiki
+//
+//  Created by Артем Маков on 15/02/2020.
+//  Copyright © 2020 Artem Makov. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController, UISearchBarDelegate {
+    
+    @IBOutlet weak var navItem: UINavigationItem!
+    @IBOutlet weak var tableView: UITableView!
+    let searchController = UISearchController(searchResultsController: nil)
+    var heroArray:[Hero] = []
+    var filteredHeroes: [Hero] = []
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        generateData()
+        registerTableView()
+        registerSearchController()
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        let titleView = UILabel(frame: CGRect(x:0, y:0, width: 40,height: 40))
+        titleView.contentMode = .scaleAspectFit
+        titleView.text = "Search"
+        navItem.titleView = titleView
+    }
+    
+    private func generateData() {
+        let deffaultHero: Hero = Hero(name: "Baba Vova", planet: "Sormovo")
+        
+        for i in 0...30 {
+            heroArray.append(Hero(name: deffaultHero.name+String(i),  planet: deffaultHero.planet+String(i),sex: getRandomSex()))
+        }
+    }
+    
+    private func registerTableView() {
+        registerTableViewCells()
+        tableView.delegate = self
+        tableView.dataSource = self
+    }
+    
+    private func registerTableViewCells() {
+        let cell = UINib(nibName: "SearchTableViewCell", bundle: nil)
+        tableView.register(cell, forCellReuseIdentifier: "SearchTableViewCell")
```

Comment:
Consider to use String(describing: SearchTableViewCell.self) when set cell id or nib name as object class' name